gimme

```diff
@@ -81,13 +81,32 @@ _do_curl() {
 	exit 1
 }
 
+# _sha256sum "file"
+_sha256sum() {
+	if command -v sha256sum &>/dev/null; then
+		sha256sum "$@"
+	else
+		shasum -a 256 "$@"
+	fi
+}
+
 # _do_curls "file" "url" ["url"...]
 _do_curls() {
 	f="${1}"
 	shift
-	[[ ! -s "${f}" ]] || return 0
+	if _sha256sum -c "${f}.sha256" &>/dev/null; then
+		return 0
+	fi
 	for url in "${@}"; do
 		if _do_curl "${url}" "${f}"; then
+			if _do_curl "${url}.sha256" "${f}.sha256"; then
+				echo "  ${f}" >>"${f}.sha256"
```

Comment:
This appears to be relying upon some undocumented mode of line continuation which I've not seen before, or upon the remote not including a terminating newline for the sha file, which seems like a dangerous assumption waiting to bite us.

```
echo "$(cat "${f}.sha256")  ${f}" > "${f}.newsha" && mv -v "${f}.newsha" "${f}.sha256"
```

(untested)